### PR TITLE
Async expect matchers

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**
Issue: https://github.com/blackbaud/skyux-sdk-testing/issues/32
Docs PR: https://github.com/blackbaud/skyux2-docs/pull/1017

The latest version of `jasmine` now supports `expectAsync` so we can write truly async matchers for `toBeAccessible`, `toEqualResourceString`, and `toContainResourceString`.

```
// Instead of...
expect(element).toBeAccessible(done);

// Do...
await expectAsync(element).toBeAccessible();
```

This also works with the `.not` operator:
```
await expectAsync(element).not.toBeAccessible();
```

- I also deprecated the existing "async" matchers `toBeAccessible`, `toEqualResourceString`, and `toContainResourceString`.

### CODE REVIEW

File: src/app/public/matchers/matchers.ts [hunk 1]
```diff
@@ -323,12 +411,40 @@ interface SkyNormalMatchers<T> extends jasmine.Matchers<T> {
...
-}
 
-/**
- * Interface for custom Sky matchers (includes original jasmine matchers).
- */
-export interface SkyMatchers<T> extends SkyNormalMatchers<T>, SkyAsyncMatchers<T> {}
+  /**
+   * `expect` the actual component to be accessible based on Web Content Accessibility
+   * Guidelines 2.0 (WCAG20) Level A and AA success criteria.
+   * @deprecated Use `expectAsync(element).toBeAccessible()` instead.
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/matchers/matchers.ts:418]: Should this example also include an `await` the proceeds the `expectAsync` ?

Comment (Blackbaud-AlexKingman) [reply src/app/public/matchers/matchers.ts:418]: Same goes for the other methods below.

**Review by Blackbaud-AlexKingman — COMMENTED** [review]
Nice work! Thanks for doing this. Just one docs question.

File: src/app/public/matchers/matchers.ts (hunk 1 above)
Comment (Blackbaud-SteveBrush) [reply src/app/public/matchers/matchers.ts]: Done.

**Review by Blackbaud-AlexKingman — APPROVED** [review]

**Pull request merged**